openmc/data/thermal.py

```diff
@@ -0,0 +1,396 @@
+            HDF5 group to read from
+
+        Returns
+        -------
+        openmc.data.ThermalScattering
+            Neutron thermal scattering data
+
+        """
+        name = group.name[1:]
+        atomic_weight_ratio = group.attrs['atomic_weight_ratio']
+        temperature = group.attrs['temperature']
+        table = cls(name, atomic_weight_ratio, temperature)
+        table.zaids = group.attrs['zaids']
+
+        # Read thermal elastic scattering
+        if 'elastic' in group:
+            elastic_group = group['elastic']
+
+            # Cross section
+            elastic_xs_type = elastic_group['xs'].attrs['type'].decode()
+            if elastic_xs_type == 'tab1':
+                table.elastic_xs = Tabulated1D.from_hdf5(elastic_group['xs'])
+            elif elastic_xs_type == 'bragg':
+                table.elastic_xs = CoherentElastic.from_hdf5(elastic_group['xs'])
+
+            # Angular distribution
+            if 'mu_out' in elastic_group:
+                table.elastic_mu_out = elastic_group['mu_out'].value
+
+        # Read thermal inelastic scattering
+        if 'inelastic' in group:
+            inelastic_group = group['inelastic']
+            table.secondary_mode = inelastic_group.attrs['secondary_mode'].decode()
+            table.inelastic_xs = Tabulated1D.from_hdf5(inelastic_group['xs'])
+            if table.secondary_mode in ('equal', 'skewed'):
+                table.inelastic_e_out = inelastic_group['energy_out']
+                table.inelastic_mu_out = inelastic_group['mu_out']
+            elif table.secondary_mode == 'continuous':
+                table.inelastic_dist = AngleEnergy.from_hdf5(inelastic_group)
+
+        return table
+
+    @classmethod
+    def from_ace(cls, ace_or_filename, name=None):
+        """Generate thermal scattering data from an ACE table
+
+        Parameters
+        ----------
+        ace : openmc.data.ace.Table or str
+            ACE table to read from. If given as a string, it is assumed to be
+            the filename for the ACE file.
+        name : str
+            GND-conforming name of the material, e.g. c_H_in_H2O. If none is
+            passed, the appropriate name is guessed based on the name of the ACE
+            table.
+
+        Returns
+        -------
+        openmc.data.ThermalScattering
+            Thermal scattering data
+
+        """
+        if isinstance(ace_or_filename, Table):
+            ace = ace_or_filename
+        else:
+            ace = get_table(ace_or_filename)
+
+        # Get new name that is GND-consistent
+        ace_name, xs = ace.name.split('.')
+        if name is None:
+            if ace_name.lower() in _THERMAL_NAMES:
+                name = _THERMAL_NAMES[ace_name.lower()] + '.' + xs
+            else:
+                # Make an educated guess?? This actually works well for JEFF-3.2
+                # which stupidly uses names like lw00.32t, lw01.32t, etc. for
+                # different temperatures
```

Comment:
Not a single nuclear data library is consistent, clean, or sane.  At least JEFF 3.2 doesn't have a freaking square wave in Fe-56.